python-package/xgboost/ml/dmlc/param/internal.py

```diff
@@ -0,0 +1,163 @@
+#
+# Copyright (c) 2022 by Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from abc import ABCMeta
+
+from pyspark.ml.classification import Classifier, ProbabilisticClassifier, \
+    ProbabilisticClassificationModel, ClassificationModel
+from pyspark.ml.common import inherit_doc
+from pyspark.ml.regression import Regressor, RegressionModel
+from pyspark.ml.util import JavaMLReadable, JavaMLReader, JavaMLWritable
+from pyspark.ml.wrapper import JavaPredictor, JavaPredictionModel
+
+from xgboost.ml.dmlc.param.shared import _XGBoostRegressorParams, _XGBoostClassifierParams
+
+
+class _XGBJavaMLReadable(JavaMLReadable):
+    """
+    Mixin a version of xgboost JavaMLReadable
+    """
+
+    @classmethod
+    def read(cls):
+        return _XGBJavaMLReader(cls)
+
+
+class _XGBJavaMLReader(JavaMLReader):
+    """
+    Mixin a version of xgboost JavaMLReader
+    """
+
+    @classmethod
+    def _java_loader_class(cls, clazz):
+        if hasattr(clazz, '_java_class_name') and clazz._java_class_name is not None:
+            return clazz._java_class_name
+        else:
+            return JavaMLReader._java_loader_class(clazz)
+
+
+@inherit_doc
+class _XGBJavaClassifier(Classifier, JavaPredictor, metaclass=ABCMeta):
+    """
+    Java Classifier for classification tasks.
+    Classes are indexed {0, 1, ..., numClasses - 1}.
+    """
+
+    # Copied from _JavaClassifier
+    def setRawPredictionCol(self, value):
+        """
+        Sets the value of :py:attr:`rawPredictionCol`.
+        """
+        return self._set(rawPredictionCol=value)
+
+
+@inherit_doc
+class _XGBJavaProbabilisticClassifier(ProbabilisticClassifier, _XGBJavaClassifier,
+                                      metaclass=ABCMeta):
+    """
+    Java Probabilistic Classifier for classification tasks.
+    """
+    pass
+
+
+class _XGBoostClassifierBase(_XGBJavaProbabilisticClassifier, _XGBoostClassifierParams,
+                             JavaMLWritable, _XGBJavaMLReadable):
+    """
+    The base class of XGBoostClassifier
+    """
+    pass
+
+
+@inherit_doc
```

Comment:
I'm not entirely sure what's the result of combining this decorator with a custom doc string.  Have you checked?